plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/config/KieBpmConfigAction.java

```diff
@@ -64,6 +62,26 @@ public String save() {
 		}
 		return SUCCESS;
 	}
+	
+	public String test() {
+		try {
+			// enable plugin by default;
+			this.setActive(true);
+			KieBpmConfig config = this.modelToConfig();
+			this.getFormManager().updateConfig(config);
+            try {
+                this.getFormManager().getContainersList();
+                this.addActionMessage(this.getText("message.config.test.success"));
+            } catch (Exception e) {
+                _logger.error("Configuration test failed!", e);
+                this.addActionError(this.getText("message.config.test.fail"));
+            }
```

Comment:
If this block executes the method will return SUCCESS. Is that the intent? Do throwable and Exception need to be handled separately like this or can we unify the separate catch blocks?

Reply:
The test is considered successful when it is possible to fetch the list of containers from the BPM calling getContainersList(). 
If the plugin is misconfigured the method above will certainly throw an exception: this exception makes the connection test fail (an the appropriate message will appear in the frontend) but this is expected to happen - the test has been executed.
So, the inner try{} is used to make the test always fail or succeed, for this reason it only guards the method getContainersList().
The outer try{} catches all the unexpected (and unlikely) exceptions which would signal a bug in the test() method itself.
Had the test() method been bigger I would have written the same come differently -no one like nested try/catch I think - but I thought that for a little method like that it wasn't necessary.
I cannot merge into a single try because other methods in test() (e.g. updateConfig) might raise the same exception and the only call that might fail is getContainersList() - everything else is a bug.

Reply:
Ah that makes sense. Thanks for the detailed explanation.